### ISSUE TITLE
[Elastic Agent] - RPM / DEB not working, Agent can't enroll to fleet - shows message: "Usage: /etc/init.d/elastic-agent {start|stop|status|restart|force-reload}"

### ISSUE THREAD

**Issue by EricDavisX** [elastic/beats#22744]
On enrolling the .deb 7.11-SNAPSHOT package agent, we are getting the following message
"Usage: /etc/init.d/elastic-agent {start|stop|status|restart|force-reload}"
and are unable to enroll the .deb agent using the following steps on Linux Debian 9 with .deb or a Centos8 with .rpm instance (verified on both, but deb example is given below:

Steps followed:

Download the "elastic-agent-7.11.0-amd64.deb" package artifact page.
Agent: https://snapshots.elastic.co/7.11.0-2a00e604/downloads/beats/elastic-agent/elastic-agent-7.11.0-SNAPSHOT-amd64.deb

place the agent package on home/zeus location on Debian 9 Linux machine.

Run the command to get the root/admin permission: sudo su root

sudo dpkg -i elastic-agent-7.11.0-SNAPSHOT-amd64.deb

sudo elastic-agent enroll https://df36b1e2a07749b8842cecf42ae7358a.us-central1.gcp.foundit.no:9243 SG5ZdS1IVUI2UFEzTEg4aEgyWGI6VEUzbnB6QjRRYWF0aW1Bbkg5XzBCUQ==

Observe "Usage: /etc/init.d/elastic-agent {start|stop|status|restart|force-reload}" message is displayed.

Execute sudo /etc/init.d/elastic-agent start
and observe that .deb agent is not displayed in UI.

screenshot
![unnamed](https://user-images.githubusercontent.com/12970373/100146867-8fdc7b80-2e68-11eb-925d-81ee21b3eea7.jpg)

**Comment by EricDavisX** [comment]
thank you @dikshachauhan-qasource @rahulgupta-qasource for bringing this issue, I logged it so it could be tracked more immediately and we can clean up the other test issue where it was first noted, https://github.com/elastic/beats/issues/22549#issuecomment-729674839

**Comment by EricDavisX** [comment]
if anything is changing / changed in our intended usage from 7.10 to 7.11 please do let us know and we can track updates to the Docs too, this page: https://www.elastic.co/guide/en/fleet/current/elastic-agent-installation.html

**Comment by EricDavisX** [comment]
I verified the same on a known working good Centos8 image - the .rpm install shows the same as .deb example.

**Comment by blakerouse** [comment]
This looks to be a machine configuration issue and not with elastic-agent itself.

I think the issue is your machine as `/etc/init.d` in the `PATH` environment variable. That should never be in your `PATH`. Remove `/etc/init.d` from your `PATH` environment variable then the `/usr/bin/elastic-agent` should be picked up.

**Comment by EricDavisX** [comment]
I'm seeing a separate error on a RHEL7 image:

[zeus@mainqa-atlcolo-10-0-7-174 elastic-agent-7.10.0-linux-x86_64]$ sudo ./elastic-agent install -f --kibana-url=https://asdf.eastus2.azure.elastic-cloud.com:443 --enrollment-token=asdf==
The Elastic Agent is currently in BETA and should not be used in production

Error: failed to install service (elastic-agent): exit status 1

And I can confirm a separate, different RHEL7 image works. : /

Note: we got the .rpm / .deb tests added back in to e2e-testing, here:
https://github.com/elastic/e2e-testing/pull/439
... and there, I'm seeing a failure.  Tho the code and centos/debian (public) images were working prior to the install feature refactor.  It may yet be that all of them have /etc/init.d in PATH.  We can cross that off the list and open individual tickets if needed, thanks for the report in Blake.  I'll reply back with more details.

**Comment by EricDavisX** [comment]
oh I'm sorry - I got confused again on what I was tracing.  The above note is mis-leading... I was testing on RHEL7 with .tar.gz package and found a case that didn't work and shouldn't have tied that to this issue.
...I'll log the above RHEL7 issue separately. : /

To clarify:
I checked the $PATH on the centos docker container that the e2e-testing system uses and it didn't have .init.d in the $PATH at all.  So, it seems a separate issue from this / that problem.  See screenshot:
<img width="666" alt="centos-docker-e2e-testing-container" src="https://user-images.githubusercontent.com/12970373/100797227-cded1880-33ef-11eb-9309-7cdd3c92581f.png">

I also tested the Centos8 image I said I had tested, adn find the same problem exists as tested in 7.10.1:
![Agent- rpm-deploy-7 10 1](https://user-images.githubusercontent.com/12970373/100802662-d21d3400-33f7-11eb-96d7-addcda4945cc.png)

@blakerouse do you want a separate issue for the e2e-testing container scenario?  Maybe not until we can figure out the issue?  I am glad that we keep finding things and fixing them for .deb/.rpm it is hard going tho.  Maybe using the e2e-testing repo test (that is currently marked and being merged as 'skipped') will allow a super quick way to test and iterate over this.

**Comment by blakerouse** [comment]
@EricDavisX can you provide the output of `which elastic-agent`?

**Comment by EricDavisX** [comment]
@blakerouse sure:
[zeus@mainqa-atlcolo-10-0-7-165 ~]$ which elastic-agent
/usr/bin/elastic-agent

**Comment by blakerouse** [comment]
@EricDavisX Then what is the output of `cat /usr/bin/elastic-agent`?

**Comment by EricDavisX** [comment]
looks like codez.  does this look right to you for version 10.0.1:

`#!/bin/bash
#
# elastic-agent          elastic-agent shipper
#
# chkconfig: 2345 98 02
# description: Starts and stops a single elastic-agent instance on this system
#

### BEGIN INIT INFO
# Provides:          elastic-agent
# Required-Start:    $local_fs $network $syslog
# Required-Stop:     $local_fs $network $syslog
# Default-Start:     2 3 4 5
# Default-Stop:      0 1 6
# Short-Description: Agent manages other beats based on configuration provided.
# Description:       Elastic-Agent is a shipper part of the Elastic Beats
#                    family. Please see: https://www.elastic.co/products/beats
### END INIT INFO



PATH=/usr/bin:/sbin:/bin:/usr/sbin
export PATH

[ -f /etc/sysconfig/elastic-agent ] && . /etc/sysconfig/elastic-agent
pidfile=${PIDFILE-/var/run/elastic-agent.pid}
agent=${BEATS_AGENT-/usr/share/elastic-agent/bin/elastic-agent}
args="-c /etc/elastic-agent/elastic-agent.yml --path.home /usr/share/elastic-agent --path.config /etc/elastic-agent --path.logs /var/log/elastic-agent"
beat_user="${BEAT_USER:-root}"
wrapper="/usr/share/elastic-agent/bin/elastic-agent-god"
wrapperopts="-r / -n -p $pidfile"
user_wrapper="su"
user_wrapperopts="$beat_user -c"
RETVAL=0

# Source function library.
. /etc/rc.d/init.d/functions

# Determine if we can use the -p option to daemon, killproc, and status.
# RHEL < 5 can't.
if status | grep -q -- '-p' 2>/dev/null; then
    daemonopts="--pidfile $pidfile"
    pidopts="-p $pidfile"
fi

if command -v runuser >/dev/null 2>&1; then
    user_wrapper="runuser"
fi

[ "$beat_user" != "root" ] && wrapperopts="$wrapperopts -u $beat_user"

start() {
    echo -n $"Starting elastic-agent: "
	if [ $? -ne 0 ]; then
		echo
		exit 1
	fi
    daemon $daemonopts $wrapper $wrapperopts -- $agent $args
    RETVAL=$?
    echo
    return $RETVAL
}

stop() {
    echo -n $"Stopping elastic-agent: "
    killproc $pidopts $wrapper
    RETVAL=$?
    echo
    [ $RETVAL = 0 ] && rm -f ${pidfile}
}

restart() {
	if [ $? -ne 0 ]; then
		return 1
	fi
    stop
    start
}

rh_status() {
    status $pidopts $wrapper
    RETVAL=$?
    return $RETVAL
}

rh_status_q() {
    rh_status >/dev/null 2>&1
}

case "$1" in
    start)
        start
    ;;
    stop)
        stop
    ;;
    restart)
        restart
    ;;
    condrestart|try-restart)
        rh_status_q || exit 0
        restart
    ;;
    status)
        rh_status
    ;;
    *)
        echo $"Usage: $0 {start|stop|status|restart|condrestart}"
        exit 1
esac

exit $RETVAL`

**Comment by blakerouse** [comment]
Okay something is wrong with the RPM package then. That is the file that should be placed in `/etc/init.d` and a smaller simple wrapper should be installed in `/usr/bin/elastic-agent`.

I never ran into this issue in my testing, something must have changed in the package.yml that broke this.

**Comment by ph** [comment]
@blakerouse Did you investige it?

**Comment by blakerouse** [comment]
@ph Yes, seems the package building is copying the wrong file to the `/usr/bin/elastic-agent` file. Still investigating why, because the `packages.yml` looks correct.

**Comment by blakerouse** [comment]
I think I figured it out. Happens that I made the files the same name, so when the package is build the `/etc/init.d` script overwrites the `/usr/bin` script. Then when the package is installed the overwritten file place it both paths.

I have a fix I am testing now, should be a simple fix, if I am correct.

**Comment by vipulsrivastava-qasource** [comment]
Hi @EricDavisX

We have performed regression testing for Linux deb/rpm on 7.10.2 BC1 Kibana cloud build. Build details is as follows:

```
Platform: Staging
Version: 7.10.2
Commit: a0b7936
Build number: 36136
Artifact: https://staging.elastic.co/7.10.2-e6b2b8c2/summary-7.10.2.html
```

We have execute test cases under test plan execution :

- [Test Run for Linux .deb deploy of BC Agent/Endpoint with minimal sanity test](https://elastic.testrail.io/index.php?/runs/view/952&group_by=cases:section_id&group_order=asc)
- [Test Run for Linux .rpm deploy of BC Agent/Endpoint with minimal sanity test](https://elastic.testrail.io/index.php?/runs/view/955&group_by=cases:section_id&group_order=asc)

Please let us know if anything else is required to be covered. 

Thanks
